projects/aws/bottlerocket-bootstrap/pkg/utils/userdata.go

```diff
@@ -24,28 +27,89 @@ type WriteFile struct {
 	Content     string
 }
 
+type UserDataSource struct {
+	Provider string
+	Prefix   string
+	Chunks   int
+}
+
+type ExternalUserData struct {
```

Comment:
Looks like this does not house the userdata itself and only the source, can we merge the above struct with this, and rename this one to something like `ExternalUserDataSource`?